crates/pixi_manifest/src/manifests/manifest.rs

```diff
@@ -531,14 +534,52 @@ impl Manifest {
         };
         let to_add: IndexSet<_> = channels.into_iter().collect();
         let new: IndexSet<_> = to_add.difference(current).cloned().collect();
+        let new_channels: IndexSet<_> = new.clone().into_iter().map(|channel| channel.channel).collect();
+
+        // clear channels with modified priority
+        current.retain(|c| ! new_channels.contains(&c.channel));
 
-        // Add the channels to the manifest
+        // Add the updated channels to the manifest
         current.extend(new.clone());
 
         // Then to the TOML document
         let channels = self.document.get_array_mut("channels", feature_name)?;
+
         for channel in new.iter() {
-            channels.push(channel.channel.to_string())
+            // lookup index for changed priority and update in-place
+            // is there an easier lookup for deserialize to use here?
+            let index = channels.iter().map(
+                |value| {
+                    let deserializer = value.to_string().trim().parse::<toml_edit::de::ValueDeserializer>().unwrap();
+                    TomlPrioritizedChannelStrOrMap::deserialize(deserializer)
+            }
+            ).position(
+                |c| match c {
+                    Ok(c) => c.into_prioritized_channel().channel == channel.channel,
+                    Err(_) => false,
+                }
+            );
+
+            // push if not found, remove & insert if found
+            let mut push = |item: Value| {
+                if index.is_some() {
+                    channels.remove(index.unwrap());
+                    channels.insert(index.unwrap(), item);
+                } else {
+                    channels.push(item);
+                }
+            };
+            match channel.priority {
+                Some(priority) => {
+                    let mut table = Table::new().into_inline_table();
+                    table.insert("channel", channel.channel.to_string().into());
+                    table.insert("priority", i64::from(priority).into());
+                    push(table.into());
```

Comment:
Instead of you could implement the `toml_edit::Value` for this `PrioritizedChannel`. Which then might be reused in different places.